forum/src/main/java/unipd/se18/ocrcamera/forum/ForumLogin.java

```diff
@@ -0,0 +1,174 @@
+package unipd.se18.ocrcamera.forum;
+
+import android.arch.lifecycle.Observer;
+import android.arch.lifecycle.ViewModelProvider;
+import android.arch.lifecycle.ViewModelProviders;
+import android.content.Context;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import unipd.se18.ocrcamera.forum.viewmodels.Login_VM;
+
+
+/**
+ * A fragment where a user can login to the forum service
+ *
+ * @author Leonardo Rossi (g2), Alberto Valente (g2), Taulant Bullaku (g2)
+ */
+public class ForumLogin extends Fragment {
+
+    /**
+     * ***************************
+     * **   GLOBAL VARIABLES    **
+     * ***************************
+     */
+
+    private OnFragmentInteractionListener mListener;
+
+    /**
+     * String used for logs to identify the fragment throwing it
+     */
+    private final String LOG_TAG = "@@ForumLogin";
+
+    /**
+     * String used to indicate that no username has been passed
+     */
+    private final String LOG_NULL_USERNAME = "A NULL username passed through liveLoginResponse";
+
+    /**
+     * String used to indicate that no error message has been passed
+     */
+    private final String LOG_NULL_ERROR_MESSAGE = "A NULL error message passed through liveError";
```

Comment:
Yes, I did it to avoid hard coding.
I preferred doing this because i thought it was more important for the purpose of cleaner and more precise code.
But thanks for the advice